src/services/dashboards/widgets/aws-cloud-front-cost/AWSCloudFrontCost.vue

```diff
@@ -39,122 +45,217 @@ import {
+    more: boolean;
+}
+type UsageType = 'data-transfer.out' | 'requests.http' | 'requests.https';
+
+const USAGE_SOURCE_UNIT = 'GB';
+const SELECTOR_TYPE_FIELDS_KEY_MAP = {
+    cost: 'usd_cost',
+    usage: 'usage_quantity',
+};
+const USAGE_TYPE_LABEL_MAP: Record<UsageType, string> = {
+    'data-transfer.out': 'Transfer-out',
+    'requests.http': 'Requests (HTTP)',
+    'requests.https': 'Requests (HTTPS)',
 };
-const CATEGORY_FIELD_NAME = GROUP_BY.PROJECT;
 
 const props = defineProps<WidgetProps>();
 const chartContext = ref<HTMLElement|null>(null);
-const {
-    createXYVerticalChart, createXYColumnSeries, createLegend,
-    createTooltip, setXYSharedTooltipText, disposeRoot, refreshRoot,
-    // setChartColors,
-} = useAmcharts5(chartContext);
+const chartHelper = useAmcharts5(chartContext);
 
 const state = reactive({
-    ...toRefs(useWidgetState<HistoryDataModel['results']>(props)),
-    labels: computed(() => ['data-transfer.out', 'requests.http', 'requests.https']),
-    tableFields: computed(() => [
-        GROUP_BY_ITEM_MAP[GROUP_BY.PROJECT],
-        { name: 'data-transfer.out', label: 'Transfer-Out' },
-        { name: 'requests.http', label: 'Requests (HTTP)' },
-        { name: 'requests.https', label: 'Requests (HTTPS)' },
-    ]),
+    ...toRefs(useWidgetState<FullData>(props)),
+    fieldsKey: computed<string>(() => SELECTOR_TYPE_FIELDS_KEY_MAP[state.selectedSelectorType]),
+    legends: [] as Legend[],
+    chartData: computed(() => {
+        const valueKey = `${state.fieldsKey}_sum`;
+        const _chartData = getRefinedXYChartData(state.data?.results, state.groupBy, 'usage_type', valueKey, true);
+        return reverse(_chartData);
+    }),
+    tableFields: computed<Field[]>(() => {
+        if (!state.groupBy) return [];
+        const textOptions: Field['textOptions'] = {
+            type: state.fieldsKey === 'usd_cost' ? 'cost' : 'size',
+            sourceUnit: USAGE_SOURCE_UNIT,
+        };
+        return [
+            GROUP_BY_ITEM_MAP[state.groupBy],
+            { name: `${state.fieldsKey}_sum.0.value`, label: 'Transfer-out', textOptions },
+            { name: `${state.fieldsKey}_sum.1.value`, label: 'Requests (HTTP)' },
+            { name: `${state.fieldsKey}_sum.2.value`, label: 'Requests (HTTPS)' },
+        ];
+    }),
+    dateRange: computed<DateRange>(() => {
+        const end = state.settings?.date_range?.end ?? dayjs.utc().format('YYYY-MM');
+        const range = props.size === WIDGET_SIZE.full ? 11 : 3;
+        const start = dayjs.utc(end).subtract(range, 'month').format('YYYY-MM');
+        return { start, end };
+    }),
+    thisPage: 1,
 });
 
 const widgetFrameProps:ComputedRef = useWidgetFrameProps(props, state);
 
 /* Api */
-const fetchData = async () => new Promise((resolve) => {
-    setTimeout(() => {
-        resolve(SAMPLE_RAW_DATA.results);
-    }, 1000);
-});
+const fetchData = async (): Promise<FullData> => {
+    if (!state.groupBy) return { results: [], more: false };
```

Comment:
question: Is this error case? If this is an error case, what about log console.error()?

Reply:
In this widget, `group_by` is required field.
When I set undefined to `state.groupBy`, it occurred 500 error!

![스크린샷 2023-01-04 오후 6 28 37](https://user-images.githubusercontent.com/18563857/210524489-9d1bdad9-ca16-4d0b-8012-f5e1ea8607ea.png)

![스크린샷 2023-01-04 오후 6 29 21](https://user-images.githubusercontent.com/18563857/210524626-b91162bf-8aa7-40c0-963e-f3b42ffa24e3.png)

Reply:
conclusion: This is error case and it needs invalid text!